docs/src/pages/guides/migration-v3/migration-v3.md

```diff
@@ -0,0 +1,192 @@
+
+  ```diff
+  -const DeepChild = withTheme()(DeepChildRaw);
+  +const DeepChild = withTheme(DeepChildRaw);
+  ```
+
+### Theme
+
+- The `theme.palette.augmentColor()` method no longer performs a side effect on its input color.
+In order to use it correctly, you have to use the output of this function.
+
+  ```diff
+  -const background = { main: color };
+  -theme.palette.augmentColor(background);
+  +const background = theme.palette.augmentColor({ main: color });
+
+  console.log({ background });
+  ```
+
+### Typography
+
+- Remove the deprecated typography variants. You can upgrade by performing the following replacements:
+  - display4 => h1
+  - display3 => h2
+  - display2 => h3
+  - display1 => h4
+  - headline => h5
+  - title => h6
+  - subheading => subtitle1
+  - body2 => body1
+  - body1 (default) => body2 (default)
+- Remove the opinionated `display: block` default typograpghy style.
+You can use the new `display?: 'initial' | 'inline' | 'block';` property.
+- Rename the `headlineMapping` property to better align with its purpose.
+
+  ```diff
+  -<MuiTypography headlineMapping={headlineMapping}>
+  +<MuiTypography variantMapping={variantMapping}>
+  ```
+
+### Button
+
+- Remove the deprecated button flat, raised and fab variants:
+
+  ```diff
+  -<Button variant="raised" />
+  +<Button variant="contained" />
+  ```
+
+  ```diff
+  -<Button variant="flat" />
+  +<Button variant="text" />
+  ```
+
+  ```diff
+  -import Button from '@material-ui/core/Button';
+  -<Button variant="fab" />
+  +import Fab from '@material-ui/core/Fab';
+  +<Fab />
+  ```
+
+### TextField
+
+- You should be able to override all the styles of the FormLabel component using the css API of the InputLabel component. We do no longer need the `FormLabelClasses` property.
+
+  ```diff
+  <InputLabel
+  - FormLabelClasses={{ asterisk: 'bar' } }
+  + classes={{ asterisk: 'bar' } }
+  >
+    Foo
+  </InputLabel>
+  ```
+
+### Grid
+
+- In order to support arbitrary spacing values and to remove the need to mentally county by 8, we are changing the spacing API:
+
+  ```diff
+    /**
+     * Defines the space between the type `item` component.
+     * It can only be used on a type `container` component.
+     */
+  -  spacing: PropTypes.oneOf([0, 8, 16, 24, 32, 40]),
+  +  spacing: PropTypes.oneOf([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+  ```
+
+  Going forward, you can use the theme to implement [a custom Grid spacing transformation function](https://material-ui.com/system/spacing/#transformation).
+
+### Table
+
+- We have removed the deprecated numeric property.
+
+  ```diff
+  -<TableCell numeric>{row.calories}</TableCell>
+  +<TableCell align="right">{row.calories}</TableCell>
+  ```
+- We have removed the fixed height property on the table row.
+    The cell height is computed by the browser using the padding and line-height.
+- The `dense` mode was promoted to a different property:
```

Comment:
Not sure if I would use the word promoted here.

Reply:
moved?

Reply:
Sounds fine.